windows/bleachbit.nsi

```diff
@@ -1,493 +1,519 @@
-    "$(ALREADY_INSTALLED)" \
-    /SD IDOK \
-    IDOK uninstall_old
-
-  Abort
-
-  uninstall_old:
-  ; If installing in silent mode, also uninstall in silent mode
-  Var /GLOBAL uninstaller_cmd
-  StrCpy $uninstaller_cmd '$R0 _?=$INSTDIR'
-  IfSilent 0 +2
-  StrCpy $uninstaller_cmd "$uninstaller_cmd /S"
-  ExecWait $uninstaller_cmd ; Actually run the uninstaller
-
-  new_install:
-
-
-FunctionEnd
-
-
-;--------------------------------
-;Uninstaller Section
-
-UninstallText "$(UNINSTALL_TEXT)"
-
-Section "$(SECTION_UNINSTALL)" SECTION_UNINSTALL
-    Delete $INSTDIR\bleachbit.exe.log
-
-    !ifndef NoTranslations
-      !include LocaleToUninstall.nsh
-    !endif
-
-    !include FilesToUninstall.nsh
-
-    DeleteRegKey HKCU "Software\${prodname}"
-    # delete normal shortcuts
-    RMDir /r "$SMPROGRAMS\${prodname}"
-    # delete any extra shortcuts
-    Delete "$DESKTOP\BleachBit.lnk"
-    Delete "$QUICKLAUNCH\BleachBit.lnk"
-    Delete "$SMSTARTUP\BleachBit.lnk"
-    # Remove Windows Explorer Shell Extension (Shredder)
-    DeleteRegKey HKCR "${SHRED_REGEX_KEY}"
-
-    # Remove the uninstaller as the very last step.
-    # If something goes wrong, let the user run it again.
-    Delete $INSTDIR\uninstall.exe
-    RMDir /REBOOTOK "$INSTDIR\."
-
-    # Remove the uninstaller from registry as the very last step.
-    # If something goes wrong, let the user run it again.
-    !insertmacro MULTIUSER_RegistryRemoveInstallInfo
-SectionEnd
-
-
-;--------------------------------
-;Uninstaller Functions
-
-Function un.onInit
-
-  !insertmacro MULTIUSER_UNINIT
-
-  !insertmacro MUI_UNGETLANGUAGE
-
-FunctionEnd
-
+;  vim: ts=4:sw=4:expandtab
+;
+;  BleachBit
+;  Copyright (C) 2008-2021 Andrew Ziem
+;  https://www.bleachbit.org
+;
+;  This program is free software: you can redistribute it and/or modify
+;  it under the terms of the GNU General Public License as published by
+;  the Free Software Foundation, either version 3 of the License, or
+;  (at your option) any later version.
+;
+;  This program is distributed in the hope that it will be useful,
+;  but WITHOUT ANY WARRANTY; without even the implied warranty of
+;  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;  GNU General Public License for more details.
+;
+;  You should have received a copy of the GNU General Public License
+;  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+;  @scriptversion v3.0.1.1429
+;  @scriptdate 2019-12-01
+
+
+;--------------------------------
+;Pack header:
+
+; Compress installer exehead with an executable compressor.
+
+; Paths should be absolute to allow building from any location.
+; Note that your executable compressor should not compress the first icon.
+
+!ifdef packhdr
+  ;Using UPX path info from setup_py2exe.py ->
+  !packhdr "$%TEMP%\exehead.tmp" '"\upx\upx.exe" -9 -q "$%TEMP%\exehead.tmp"'
+!endif
+
+
+;--------------------------------
+;Include Modern UI
+
+  !include "MUI2.nsh"

```

Comment:
@bovirus 

It looks like the _whole file_ was deleted and replaced, so it makes it hard to see where the differences are. Normally we can see only the lines that were changed.

The way it looks here  can be caused by changing the end of line characters (CR/LF). Less often, it is caused by changing whitespace.

Please be careful with the editor, so the changes are easier to see. This will make it easier to review and faster to get merged.

Reply:
@az0

I search draft option but I didn't find it.
Is it possible to download the temporary installer created? How do that?